crypto/src/signature.rs

```diff
@@ -263,10 +263,71 @@ impl<T: Encode> SignatureOf<T> {
     }
 }
 
+/// Wrapper around [`SignatureOf`] used to reimplement [`Eq`], [`Ord`]
+/// to compare signatures only by there [`PublicKey`].
+#[allow(clippy::derive_hash_xor_eq)]
```

Comment:
Clippy has a point. To me it's not obvious why comparing by public key in `PartialEq` is compatible with the default `hash` implementation which also includes the `payload`. This will lead to inconsistencies in behaviour in any `HashMap`, so I'd rather lay it out explicitly via a manual implementation or a unit test.

Reply:
Specifically, if we want to `impl Hash`, we need to explain why the usage as is, will not cause any problems.

Reply:
The unit test should verify that the difference in behaviour between `PartialEq` and `Hash` is there, and explain why it's OK.

Reply:
Added manual implementation and covered it with test.